Source/SuperLinq.Async/HasDuplicates.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Xml.Linq;
+
+namespace SuperLinq.Async;
+
+public static partial class AsyncSuperEnumerable
+{
+	/// <summary>
+	///   Checks if sequence contains duplicates
+	/// </summary>
+	/// <param name="source">The sequence to check.</param>
+	/// <typeparam name="T">The type of the elements in the source sequence</typeparam>
+	/// <returns>
+	/// <see langword="true"/> if any element of the sequence <paramref name="source" /> is duplicated, <see langword="false"/> otherwise 
+	/// </returns>
+	public static ValueTask<bool> HasDuplicates<T>(this IAsyncEnumerable<T> source)
+		=> source?.HasDuplicates(EqualityComparer<T>.Default) ?? throw new ArgumentNullException(nameof(source));
```

Comment:
`SuperLinq` does not allow `throw` expressions, as they reduce the ability of the JIT to inline methods, especially ones that proxy to another overload. There is no reason to use `throw` here, as the other overload will protect it via `Guard.IsNotNull()`.